crates/codegen/language/definition/src/model/item.rs

```diff
@@ -39,4 +39,16 @@ impl Item {
             Item::Fragment { item } => &item.name,
         }
     }
+
+    pub fn is_terminal(&self) -> bool {
```

Comment:
I believe `Item::Fragment` is neither a terminal or a non-terminal, so it should return `false` in both functions here. Looking at the currently only callsite, I think it is also excluded for the same reason.
I wonder why it is not the case here?

Reply:
I guess that's one instance of where I consider 'terminality' of items being a bit of a misnomer if applied too broadly. You're right that it's not a terminal in terms of the scanner structure but it does represent an item that defines how we lex and analyze (syntax tree) terminals only. It's excluded from contributing its own `TerminalKind` in that callsite but if there was a code that just collects items related to terminals, `Item::Fragment` should belong to that category IMO.

If you prefer, I can apply the suggestion and we can revisit in the future if there will be some code that might benefit from the partition logic as it currently is in the PR.

Reply:
Done in 44b4ac5fc62e9a28aa194e1680543b22fa903c80

Reply:
Sounds good. Thanks!
I think from the point of view of our parser/syntax (and coincidentally with the new renaming), terminals and nonterminals are defined as nodes in the tree. Fragments are just a DSL helper utility that allows us to refactor common scanners, but they are not nodes.
Happy to revisit this if you see otherwise 👍  just a suggestion.